plugins/exporter.koplugin/main.lua

```diff
@@ -1,431 +1,129 @@
-                            auth_dialog = InputDialog:new{
-                                title = _("Set authorization token for Readwise"),
-                                input = self.readwise_token,
-                                buttons = {
-                                    {
-                                        {
-                                            text = _("Cancel"),
-                                            id = "close",
-                                            callback = function()
-                                                UIManager:close(auth_dialog)
-                                            end
-                                        },
-                                        {
-                                            text = _("Set token"),
-                                            callback = function()
-                                                self.readwise_token = auth_dialog:getInputText()
-                                                self:saveSettings()
-                                                UIManager:close(auth_dialog)
-                                            end
-                                        }
-                                    }
-                                }
-                            }
-                            UIManager:show(auth_dialog)
-                            auth_dialog:onShowKeyboard()
-                        end
-                    },
-                    {
-                        text = _("Export to Readwise"),
-                        checked_func = function() return self.readwise_export end,
-                        callback = function()
-                            self.readwise_export = not self.readwise_export
-                            if self.readwise_export then
-                                self.html_export = false
-                                self.txt_export = false
-                                self.json_export = false
-                                self.joplin_export = false
-                            end
-                            self:saveSettings()
-                        end
-                    },
-                    {
-                        text = _("Help"),
-                        keep_menu_open = true,
-                        callback = function()
-                            UIManager:show(InfoMessage:new{
-                                text = T(_([[You can enter your auth token on your computer by saving an empty token. Then quit KOReader, edit the exporter.readwise_token field in %1/settings.reader.lua after creating a backup, and restart KOReader once you're done.
-
-For more information, please visit https://github.com/koreader/koreader/wiki/Highlight-export.]])
-                            , BD.dirpath(DataStorage:getDataDir()))
-                            })
-                        end
-                    }
-                }
+function Exporter:migrateSettings()
```

Comment:
please use a local function instead.

Also I think the code can be slimmed with something like

```lua
local function migrateSettings()
    local formats = {"html", "joplin", "json", "readwise", "text"}

    local settings = G_reader_settings:readSetting("exporter")
    if not settings then
        settings = G_reader_settings:readSetting("evernote")
    end

    if type(settings) == "table" then
        for _, fmt in ipairs(formats) do
            if type(settings[fmt]) == "table" then return end
        end
        local new_settings = {}
        for _, fmt in ipairs(formats) do
            new_settings[fmt] = { enabled = false }
        end
        new_settings["joplin"].ip = settings.joplin_IP
        new_settings["joplin"].port = settings.joplin_port
        new_settings["joplin"].token = settings.joplin_token
        new_settings["readwise"].token = settings.readwise_token
        G_reader_settings:saveSetting("exporter", new_settings)
    end
end
```

Reply:
Done.